contracts/WildcardSteward_v3.sol

```diff
@@ -534,25 +564,30 @@ contract WildcardSteward_v3 is Initializable {
                 _foreclose(tokenId);
 
                 address benefactor = benefactors[tokenId];
-
-                // If the organisation collected their patronage after this token was foreclosed, then record the credit they have been given.
-                if (
-                    timeLastCollectedBenefactor[benefactor] >
-                    newTimeLastCollected
-                ) {
-                    benefactorCredit[benefactor] = benefactorCredit[benefactor]
-                        .add(
-                        price[tokenId]
-                            .mul(
-                            (
-                                timeLastCollectedBenefactor[benefactor].sub(
-                                    newTimeLastCollected
-                                )
-                            )
+                uint256 amountOverCredited = price[tokenId]
+                    .mul(
+                    (
+                        timeLastCollectedBenefactor[benefactor].sub(
+                            newTimeLastCollected
                         )
-                            .mul(patronageNumerator[tokenId])
-                            .div(31536000000000000000) // 365 days * 1000000000000
-                    );
+                    )
+                )
+                    .mul(patronageNumerator[tokenId])
+                    .div(31536000000000000000);
+
+                if (benefactorFunds[benefactor] > 0) {
```

Comment:
This essentially nets debit and credit out and ensures based on the overdrawn amount, the correct final debit or credit value will reflect.